### PR TITLE
Memory validation

### PR DESCRIPTION
**Pull request by aalekseev-ru**
Add VCachedMemoryRequest/VCachedMemoryResponse Validate functions
I made it looking at VObjectTranscriptReport Validate function and find some redundant fields, remove it

### CODE REVIEW

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+func (m *VCachedMemoryResponse) Validate(currentPulse PulseNumber) error {
+	object := m.Object.GetValue()
+	_, err := validSelfScopedGlobalWithPulseBeforeOrEq(object, currentPulse, "Object")
+	if err != nil {
+		return err
+	}
+
+	switch m.CallStatus {
+	case CachedMemoryStateUnknown:
+		if m.Object.IsEmpty() {
```
Comment (bigbes) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: duplicated check

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			return throw.New("Object should not be empty")
+		}
+		if m.StateID.IsEmpty() {
```
Comment (bigbes) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:24]: duplicated check, can be extracted to upper level

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			return throw.New("StateID should not be empty")
+		}
+	case CachedMemoryStateFound:
+		if m.Object.IsEmpty() {
+			return throw.New("Object should not be empty")
+		}
+		if m.StateID.IsEmpty() {
+			return throw.New("StateID should not be empty")
+		}
+		if m.Class.IsEmpty() {
```
Comment (bigbes) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:34]: validSelfScopedGlobalWithPulseBeforeOrEq

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go (hunk 1 above)
Comment (aalekseev-ru) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: moved to another PR https://github.com/insolar/assured-ledger/pull/832/files#diff-27ea9ed0019287b5fd67549774d23164R20

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go (hunk 3 above)
Comment (aalekseev-ru) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:34]: moved to another PR, m.Class removed

**Pull request closed without merging**